state/internal_test.go

```diff
@@ -66,6 +73,9 @@ func (s *internalStateSuite) SetUpTest(c *gc.C) {
 		},
 		MongoInfo:     info,
 		MongoDialOpts: mongotest.DialOpts(),
+		NewPolicy: func(*State) Policy {
```

Comment:
I don't like it either, but I don't think we can without more drastic changes.

We need the registry in the policy, because we need to use the registry when adding storage dynamically. We can't use the policy when *constructing* a model, though, because policy requires access to a fully-fledged State.

I wonder if we shouldn't just have state depend on the environs package, and then State can be responsible for constructing and using an Environ directly for validation and so on. No need for a "policy" at all then.